### DISCUSSION TITLE
Suggestions, Questions, and Feature Requests for Fast Music Remover

### DISCUSSION THREAD

**Discussion by omeryusufyagci** [omeryusufyagci/fast-music-remover#6] [General]
I'd love to get your feedback on Fast Music Remover, a tool being built truly with users like you in mind! 

#### Here's some context on where we are and what we want to achieve in the near future:
We aim to provide an efficient core ([`MediaProcessor`](https://github.com/omeryusufyagci/fast-music-remover/tree/main/MediaProcessor)), that supports multiple ML models, and other modules, e.g. DSP techniques, that can address your needs and pain points in processing media files. 

We've started with `DeepFilterNet` as our initial model, which helps us achieve efficient speech isolation. In the future you can expect to see support for more models, such as `demucs`, to provide a better coverage of potential use cases. These models and techniques all have their strengths and our vision for our core is to enable you with all the tools you may need!

We're focused on building an efficient core with a clean, modular API. Our goal is to make it easy for developers to integrate `MediaProcessor` into their projects. While the core currently exists in application form for CLI convenience and testing, the processing API is already designed to function as a library.
In the future, we plan to introduce a dedicated CLI tool and provide `MediaProcessor` as a standalone library for broader use. Stay tuned!

We're also developing a project launcher that will take care of installing dependencies, configuring project files with respect to your platform, and providing a streamlined way to access our core features. With the web application and [docker image](https://github.com/omeryusufyagci/fast-music-remover/pkgs/container/fast-music-remover) we're providing, you can seamlessly use Fast Music Remover on your local machine or remote servers! 

Aiming to establish a stable release soon, we'll then take on realtime processing to support live feeds as well as the integration of other models and modules for our core. 

If you have any use cases that we're not supporting at the moment, or have any feedback really, please don't hesitate to get in touch, here or on [discord](https://discord.com/invite/xje3PQTEYp)!

**Comment by passes0d** [comment]
Can this be used to extract the music into its own file? How does it work with music that contains vocals?

**Reply by omeryusufyagci** [reply]
Hi @passes0d! The main goal of this project is to clean media of noise, music, and effects. For this purpose, we're using `DeepFilterNet`, which isn't designed for extracting instrumentals. For extracting individual instruments, I recommend looking into `demucs`-based projects, as that model is more suited for this purpose.

Regarding music as input: you may need to adjust DeepFilterNet's parameters for filtering aggressiveness. Our default values are optimized for mainstream media, but they may not work as well for songs directly. I plan to introduce a higher-level setting for dynamically adjusting filter aggressiveness, but it's not available yet.

Thanks for your interest!

**Comment by rljacobson** [comment]
This feature request is probably out of scope for this project, but I would _love_ for someone to develop a tool to "upscale" old recordings. How cool would it be if we could make old radio programs like [Yours Truly, Johnny Dollar](https://archive.org/details/OTRR_YoursTrulyJohnnyDollar_Singles) and audio from old films sound like modern studio recordings?

**Reply by omeryusufyagci** [reply]
Hi @rljacobson, thanks for bringing this up; it's always great to hear about interesting use cases from the community!

You're right, this would be something to explore after we've established our stable release. That said, I’d be open to supporting an upscaling module in the future. My vision for the core is to support various models and modules, to make it easy for users to choose the right features based on their needs. While we still have a long way to go, we’re making steady progress !

As our hands are full at the moment, I can't take this on immediately, but if there is interest from the community in contributing an upscaling module to [`MediaProcessor`](https://github.com/omeryusufyagci/fast-music-remover/tree/main/MediaProcessor) I'd be happy to support it as best as I can. 

Thanks again for sharing this idea!

**Comment by eyewee** [comment]
Hello, are there any ways to remove just music (instrumental sound) and not the special effects ? (like animal noises, rain sound etc)?

**Reply by omeryusufyagci** [reply]
Hi @eyewee, thanks for getting in touch and your interest in our project ! 

The default parameters you're using now are selected for maximum attenuation. It might get a bit better if you were to use a less aggressive value for the filtering, and quite soon I will expose this on the UI so it can be easily selected by end users. The support is already available in the core though -in case you want to hack at it in the meantime... 

So, currently, your best bet would be to play with [filter_attenuation_limit](https://github.com/omeryusufyagci/fast-music-remover/blob/966357813e16b5f3a2c00fbcaf032dcd4c1d559e/config.json#L11) and try to reduce it a bit to see if it helps. 

The model we're using right now is intended to **rapidly** isolate vocals and it isn't really meant to perform this distinction between music and environmental sounds. This was a conscious choice, as we certainly want to support live feeds in the future. However, at the same time my vision is to support a set of models that would cater to different use cases. That will come with a performance overhead, and it will be up to the user to define their requirements. 

To answer your question, it's not exactly here yet, but definitely something that's inline with the vision of the project. So, definitely stay tuned for updates! I hope this helps, and please don't hesitate to get in touch again! (we even have a [discord](https://discord.com/invite/xje3PQTEYp) if you prefer)

**Reply by sukaha** [reply]
> Hello, are there any ways to remove just music (instrumental sound) and not the special effects ? (like animal noises, rain sound etc)?

As far as I know, yes — most tools just split vocals and instrumentals, so effects like rain or animal sounds often get lost with the music. Lyruno, more specifically [lyrunocdx ](https://lyruno.com)(not lyrunomdx), does a better job here since it can separate the music while keeping those special effects intact

**Comment by HubKing** [comment]
I really truly do hate background music on YouTube videos and podcasts. Why do they keep adding loud music to their videos and audios? I haven't tested this library yet, but I did watch the demo, and it seems to be able to do what I want. Thanks in advance.

**Reply by omeryusufyagci** [reply]
Hey @HubKing, I'm glad to hear this will help you take control of the media you consume! Please let us know if you have any further feedback, and I hope you get some good use out of it! : ))

**Comment by ilyasben26** [comment]
This repo is a life-saver!! Have you thought of using an LADSPA libdeepnet filter for the real-time part? I managed to make it work on Linux so far by following the tutorial on the orignial repo (https://github.com/Rikorose/DeepFilterNet/blob/main/ladspa/README.md). But on my main Mac M1 machine, it just doesn't work. I tried compiling the plugin and then using Audacity to use it, but it simply doesn't find it. Would you happen to know any other way to make this libdeepnet processing happen in real-time (preferably cross platform)?

**Comment by AceCase499** [comment]
Hello, I am having some issues running this code and I need some help!
I was able to build a local image using Docker, and that works well.  However when I try to set up the project manually with VScode, I get an error.  Apparently an important library is missing or can't be found, even though I didn't manipulate any of the files.  Here is the error:

> ERROR:root:MediaProcessor stderr: dyld[20456]: Library not loaded: /Users/oyagci/fast-music-remover/DeepFilterNet/target/release/deps/libdf.dylib
>   Referenced from: <3B561D3B-DA00-3DEF-8F17-FEEA8DF30CEC> /Users/jerrodgraham/Documents/fast-music-remover-main/MediaProcessor/build/MediaProcessor
>   Reason: tried: '/Users/oyagci/fast-music-remover/DeepFilterNet/target/release/deps/libdf.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/oyagci/fast-music-remover/DeepFilterNet/target/release/deps/libdf.dylib' (no such file), '/Users/oyagci/fast-music-remover/DeepFilterNet/target/release/deps/libdf.dylib' (no such file)
> 
> ERROR:root:MediaProcessor returned a non-zero exit code.
> 127.0.0.1 - - [06/May/2025 06:21:28] "POST / HTTP/1.1" 200 -
> INFO:werkzeug:127.0.0.1 - - [06/May/2025 06:21:28] "POST / HTTP/1.1" 200 -

Here are the steps I took to install, please let me know if I missed a step or two:

1. Installed every dependency mentioned in the readme file
2. Downloaded the repo in a zip file
3. Unzipped and opened the project in VScode
4. Created a Python virtual environment with Python 3.13.3 64-bit (/usr/local/bin/python3)
5.  Ran 'pip install -r requirements.txt'
6. Ran the rest of the commands from the readme file:
cd MediaProcessor
mkdir build
cd build
cmake ..
make
7. Ran python3 app.py, opened in Safari and processed the demo video.  The dialogue says "Failed to process video"
I tried Google Chrome as well

What should I do differently while setting up the project?

**Reply by nimiai** [reply]
maybe you can copy fast-music-remover/MediaProcessor/lib/libdf.dylib to /Users/oyagci/fast-music-remover/DeepFilterNet/target/release/deps/

**Comment by nimiai** [comment]
Hello, can this software be used in an Android environment?

**Reply by nimiai** [reply]
can [libdf-linux-arm64.so](https://github.com/omeryusufyagci/fast-music-remover/blob/main/MediaProcessor/lib/libdf-linux-arm64.so) be used in Android Native ?

**Comment by nimiai** [comment]
你好，我是艾春玲。您的邮件已收到，我会尽快回复。
